docs/documentation/guides/adding-a-new-payment-processor.md

```diff
@@ -0,0 +1,447 @@
+        """
+            :param transaction: A Silver Transaction object in pending state, that hasn't been 
+            executed before.
+
+            Creates a real, external transaction based on the given Silver transaction.
+
+            Warning: You should never call this method directly! Use the `process_transaction` 
+                     method instead, which will call this method.
+                     However, if you need to call it directly, make sure the transaction hasn't been 
+                     executed before.
+
+            :return: True on success, False on failure.
+        """
+
+        # The following code is purely educational. Your code will probably look a lot different.
+        # Start by creating your external transaction. You usually want to specify as many details
+        # about the transaction as possible.
+        transaction_result = payment_processor_sdk.create_transaction({
+            'amount': transaction.amount,
+            'currency': transaction.currency,
+            'reference': str(transaction.uuid),
+            # 'token': transaction.payment_method.token -- We'll discuss about this later.
+        })
+
+        # Most payment processors don't necessarily execute (authorize, submit for settlement...) the
+        # transaction on their own, when creating the transaction. Let's do that:
+        transaction_result = payment_processor_sdk.execute(transaction_result.id)
+
+        # Update the Silver transaction to reflect the external's transaction new status
+        self._update_transaction_status(transaction, transaction_result)
+
+    def _update_transaction_status(self, transaction, transaction_result):
+        """
+            :param transaction: A silver transaction.
+            :param transaction_response: A response object from the payment processor SDK.
+            This method is not required by silver.
+        """
+
+        # You can do an additional check to make sure the two objects correlate:
+        if not str(transaction.uuid) == transaction_result.reference:
+            raise ValueError("The transaction and transaction_response don't match.")
+
+        # First we update the Silver transaction's external reference and status
+        transaction.external_reference = transaction_result.id
+        status = transaction_result.status
+
+        transaction.data.update({
+            'status': status,
+            'id': transaction_result.id
+        })
+        
+        # Then we transition the Silver transaction to a new state if needed:
+        if status == 'settled':
+            transaction.settle()
+        elif status == 'failed':
+            transaction.fail()
+        elif status == 'canceled':  # sometimes called voided
+            transaction.cancel()
+        elif status == 'refunded':
+            transaction.refund()
+
+        transaction.save()
+```
+
+You can specify codes and reason messages for the sad path transitions (fail, cancel, refund).
```

Comment:
`for the sad path transitions`, maybe something not that romantic: `for failing transitions`

Reply:
`error codes` and maybe a link to them would be useful.

Reply:
"happy path" and "sad path" are used often to refer to test cases, they're not meant to sound romantic. [ref](https://en.wikipedia.org/wiki/Happy_path)

Cancel and refund codes are not really error codes, because usually there's a valid reason behind those operations.  
I thought about the link to the various codes, but then I couldn't find them anywhere in the documentation. I'm pondering whether that's a different issue or not.